### PR TITLE
Profiles: Checkpoint 2

### PR DESCRIPTION
**Pull request by benisgold**
This PR is an aggregation of the following PRs:

- #3495 
- #3551 
- #3525 
- #3541 
- #3575
- #3605
- #3666 

#### Real time records validation #3495

- Ensure that field validation happens shortly after the user types, rather than at submit time

#### Support emoji avatars for ens owners w/o ens avatar #3525

- Ensure that the button on Profiles intro screen should say "Edit {ENS name}" and not "Use" if the user has a profile
- Ensure that a user with an ENS profile (but no avatar) can still pick an emoji & color

#### Fix nft upload flash when editing profile #3541

- Ensure that the ENS avatar doesn't flash shortly after uploading a photo on the Edit Profile screen.

#### Remove NFT avatar/cover option if no NFTs #3575

- Ensure that there is no option to remove avatar/cover photo if the user does not have an avatar/cover photo

#### poaps #3666

- Ensure that POAPs display on a profile that has POAPs

### CODE REVIEW

**Review by pugson — APPROVED** [review]
🔥

File: src/components/ens-registration/RegistrationAvatar/RegistrationAvatar.tsx [hunk 1]
```diff
@@ -186,7 +184,9 @@ const RegistrationAvatar = ({
                 ? onShowExplainSheet
                 : isTesting
                 ? handleSelectNFT
-                : undefined
+                : enableNFTs
+                ? undefined
+                : handleSelectImage
```
Comment (jkadamczyk) [inline src/components/ens-registration/RegistrationAvatar/RegistrationAvatar.tsx:189]: I don't like that nested ternary, any chence of moving it somewhere else or making a wrapper handler that picks what to call with if else statements?

File: src/components/ens-registration/RegistrationCover/RegistrationCover.tsx [hunk 2]
```diff
@@ -125,11 +133,17 @@ const RegistrationCover = ({
...
     >
       <ButtonPressAnimation
-        onPress={!hasSeenExplainSheet ? onShowExplainSheet : undefined}
+        onPress={
+          !hasSeenExplainSheet
+            ? onShowExplainSheet
+            : enableNFTs
+            ? undefined
+            : handleSelectImage
+        }
```
Comment (jkadamczyk) [inline src/components/ens-registration/RegistrationCover/RegistrationCover.tsx:146]: Not a fan of a nested ternary

File: src/screens/ENSIntroSheet.tsx [hunk 3]
```diff
@@ -65,33 +71,13 @@ export default function ENSIntroSheet() {
...
-      prefetchENSRecords(uniqueDomain.name);
-    }
-  }, [uniqueDomain]);
+  const profileExists = useMemo(
+    () =>
+      Object.keys(ensRecords?.coinAddresses || {}).length > 1 ||
+      ensAvatar?.imageUrl ||
+      Object.keys(ensRecords?.records || {}).length > 0,
+    [ensAvatar?.imageUrl, ensRecords?.coinAddresses, ensRecords?.records]
+  );
```
Comment (jkadamczyk) [inline src/screens/ENSIntroSheet.tsx:80]: Probably doesn't need a memo. Isn't Object.keys pretty cheap operation?

Comment (jkadamczyk) [reply src/screens/ENSIntroSheet.tsx:80]: It seems to be linear https://stackoverflow.com/questions/7716812/object-keys-complexity
Not free, but perhaps not necessary to memoize unless it re-renders a ton

File: src/screens/ProfileSheet.tsx [hunk 4]
```diff
@@ -104,7 +105,10 @@ export default function ProfileSheet() {
         <AccentColorProvider color={accentColor}>
           <Box background="body">
             <Box style={wrapperStyle}>
-              {!isPreview && (!isAddressSuccess || !hasListFetched) ? (
+              {!isPreview &&
+              (!isAddressSuccess ||
+                !hasListFetched ||
+                !hasFirstTxTimestampFetched) ? (
```
Comment (jkadamczyk) [inline src/screens/ProfileSheet.tsx:111]: May make sense to extract that boolean expression to a variable and make it somewhat descriptive like `shouldShowFoo`

**Review by jkadamczyk — APPROVED** [review]
Nits mostly

**Pull request merged**